### PR TITLE
Fix NuGet/Home#1379: VS 2015 produces incorrect binding redirects on …

### PR DESCRIPTION
**Pull request by feiling**
…update all on a VS 2013 MVC app.

The cause of the issue is that there are multiple web.config files in the project and nuget picked the
wrong one when it adds binding redirects.

The fix is to pick the one directly under the project.

@yishaigalatzer @MeniZalzman @deepakaravindr @emgarten @zhili1208

### CODE REVIEW

File: src/PackageManagement.VisualStudio/Runtime/BindingRedirectManager.cs [hunk 1]
```diff
@@ -111,7 +111,7 @@ private string GetConfigurationFileFullPath()
             var fullPaths = MSBuildNuGetProjectSystem.GetFullPaths(ConfigurationFile);
             if (!fullPaths.IsEmpty())
             {
-                return fullPaths.Last();
+                return fullPaths.First();
```
Comment (feiling) [inline src/PackageManagement.VisualStudio/Runtime/BindingRedirectManager.cs:116]: This is the fix. 

Documentation added to VSMSBuildNuGetProjectSystem.GetFullPaths() to make it clear that the first item is the app.config/web.config directly under project (if it exists).

File: test/EndToEnd/tests/install.ps1 [hunk 2]
```diff
@@ -588,6 +588,14 @@ function Test-SimpleBindingRedirects {
...
     # Arrange
     $a = New-WebApplication
+
+    # Add a another web.config under directory test
+    $testDirectory = Join-Path (Get-ProjectDir $a) "test"
+    $file = Join-Path $testDirectory "web.config"	
+    New-Item $testDirectory -ItemType Directory
+    "<configuration></configuration>" > $file
+    $a.ProjectItems.AddFromFile($file)
+
```
Comment (feiling) [inline test/EndToEnd/tests/install.ps1:598]: This test is updated to test the fix.

File: src/PackageManagement.VisualStudio/Runtime/BindingRedirectManager.cs (hunk 1 above)
Comment (emgarten) [inline src/PackageManagement.VisualStudio/Runtime/BindingRedirectManager.cs:116]: At a comment with the reason here.

File: src/PackageManagement.VisualStudio/ProjectSystems/VSMSBuildNuGetProjectSystem.cs [hunk 3]
```diff
@@ -796,7 +796,12 @@ public IEnumerable<string> GetFiles(string path, string filter, bool recursive)
                         select p.Name;
                 });
         }
-        
+
+        /// <summary>
+        /// Returns the list of full paths of all files in the project whose name is <paramref name="fileName"/>. DFS algorithm is used. Thus, the files directly under the project are returned first. Then files one-level deep are returned, and so-on.
+        /// </summary>
+        /// <param name="fileName"></param>
```
Comment (emgarten) [inline src/PackageManagement.VisualStudio/ProjectSystems/VSMSBuildNuGetProjectSystem.cs]: Add a description or remove the empty doc tags.

File: src/PackageManagement.VisualStudio/ProjectSystems/VSMSBuildNuGetProjectSystem.cs [hunk 4]
```diff
@@ -796,7 +796,12 @@ public IEnumerable<string> GetFiles(string path, string filter, bool recursive)
                         select p.Name;
                 });
         }
-        
+
+        /// <summary>
+        /// Returns the list of full paths of all files in the project whose name is <paramref name="fileName"/>. DFS algorithm is used. Thus, the files directly under the project are returned first. Then files one-level deep are returned, and so-on.
```
Comment (emgarten) [inline src/PackageManagement.VisualStudio/ProjectSystems/VSMSBuildNuGetProjectSystem.cs]: Line length

File: src/PackageManagement.VisualStudio/Runtime/BindingRedirectManager.cs [hunk 5]
```diff
@@ -111,7 +111,7 @@ private string GetConfigurationFileFullPath()
             var fullPaths = MSBuildNuGetProjectSystem.GetFullPaths(ConfigurationFile);
             if (!fullPaths.IsEmpty())
```
Comment (emgarten) [inline src/PackageManagement.VisualStudio/Runtime/BindingRedirectManager.cs]: IsEmpty() is from NuGet.Core, just use .Any()

**Comment by yishaigalatzer** [comment]
:shipit: need to add a test to validate traversal order of files

**Comment by emgarten** [comment]
:shipit:

**Comment by feiling** [comment]
Merged. Will add the test to validate traversal order of files in another pull request.

**Pull request closed without merging**